index.js

```diff
@@ -120,13 +119,32 @@ internals.levels = ['trace', 'debug', 'info', 'warn', 'error']
 
 internals.outputs = ['json', 'pretty']
 internals.output = 'json'
-internals.level = 'info'
-internals.namespaces = ''
-internals.enabledNamespaces = []
-internals.disabledNamespaces = []
+internals.level = _.indexOf(internals.level, 'info')
+internals.namespaces = []
 internals.globalContext = {}
 
 /**
+ * Parse a namespace to extract level, state and namespace (eg: info@ns1:subns1)
+ * @param {string} namespace
+ * @return {{enabled: boolean, level: number|null, regex: string|null}|null}
+ */
+internals.parseNamespace = function(namespace) {
+    const matches = /((.*)@)?(-)?(.*)/.exec(namespace)
```

Comment:
we should not allow having both level & `-` as it doesn't make sense to disable using a specific level